From 16c75423c29792209bfc638eaf4bd1f3faf89a44 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=8D=C3=B1igo=20Huguet?= <ihuguet@redhat.com>
Date: Fri, 21 Jan 2022 08:49:01 +0100
Subject: [PATCH 03/36] rtw89: Remove redundant check of ret after call to
 rtw89_mac_enable_bb_rf
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Bugzilla: http://bugzilla.redhat.com/2033291

commit f7e7e440550b0b176df3d2ea3e76106bc89915d9
Author: Colin Ian King <colin.king@canonical.com>
Date:   Fri Oct 15 16:21:13 2021 +0100

    rtw89: Remove redundant check of ret after call to rtw89_mac_enable_bb_rf

    The function rtw89_mac_enable_bb_rf is a void return type, so there is
    no return error code to ret, so the following check for an error in ret
    is redundant dead code and can be removed.

    Addresses-Coverity: ("Logically dead code")
    Fixes: e3ec7017f6a2 ("rtw89: add Realtek 802.11ax driver")
    Signed-off-by: Colin Ian King <colin.king@canonical.com>
    Acked-by: Ping-Ke Shih <pkshih@realtek.com>
    Signed-off-by: Kalle Valo <kvalo@codeaurora.org>
    Link: https://lore.kernel.org/r/20211015152113.33179-1-colin.king@canonical.com

Signed-off-by: Íñigo Huguet <ihuguet@redhat.com>
---
 drivers/net/wireless/realtek/rtw89/mac.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/net/wireless/realtek/rtw89/mac.c b/drivers/net/wireless/realtek/rtw89/mac.c
index 0171a5a7b1de..69384c43c046 100644
--- a/drivers/net/wireless/realtek/rtw89/mac.c
+++ b/drivers/net/wireless/realtek/rtw89/mac.c
@@ -2656,8 +2656,6 @@ int rtw89_mac_init(struct rtw89_dev *rtwdev)
 		goto fail;
 
 	rtw89_mac_enable_bb_rf(rtwdev);
-	if (ret)
-		goto fail;
 
 	ret = rtw89_mac_sys_init(rtwdev);
 	if (ret)
-- 
2.13.6

